File: public/UserList.php
<?php

include 'UserDao.php';

class UserList
{
  public $userList = array();

  // DBのデータをChargeクラスのリストにロード
  function __construct($squadId) {
    $userDao = new UserDao();
    $dbUsers = $userDao->get($squadId);
    $this->userList = $this->parce($dbUsers);
  }

  // ユーザー数を返却
  function userNum() {
    return count($this->userList);
  }

  // line表示用文字列に変換
  function display() {
  	$str = "";
  	foreach($this->userList as $user) {
  		$str .= $user->display();
  	}
  	return $str;
  }

  //ユーザーID一覧を配列で返却する
  function getIdArray() {
    $userIds = array();
    foreach ($this->userList as $user) {
      array_push($userIds, $user->id);
    }
    return $userIds;
  }

  function getNameArray() {
    $userNames = array();
    foreach ($this->userList as $user) {
      array_push($userNames, $user->name);
    }
    return $userNames;
  }

  // userNameが見つからなかったときは、userIdをそのまま返します。
  function getNameById($userId) {
    foreach ($this->userList as $user) {
      if ($user->id == $userId) {
        return $user->name;
      }
    }
    return $userId;
  }

  // DBデータUserクラスにパース
  private function parce($dbUsers) {
  	$users = array();
  	foreach ($dbUsers as $dbUser) {
			array_push($users, new User($dbUser['user_id'], $dbUser['user_name']));
  	}
  	return $users;
  }

}

File: UserList.rb
require_relative './User'

class UserList
    attr_reader :userList

    def initialize(userList)
        @userList = userList
    end

    # line表示用文字列に変換
    def display
        ret = ""
        @userList.each do |user|
            ret += user.display
        end
        return ret
    end

    # ユーザーID一覧を配列で返却する
    def getIdArray
        userIds = []
        @userList.each do |user|
            userIds.push(user.userId)
        end
        return userIds
    end

    def getNameArray
        userNames = []
        @userList.each do |user|
            userNames.push(user.userName)
        end
        return userNames
    end

    # userNameが見つからなかったときは、userIdをそのまま返します。
    def getNameById(userId)
        @userList.each do |user|
            return user.userName if userId == user.userId
        end
        return userId
    end

    def isAlreadyJoin(userId)
        return getIdArray().include?(userId)
    end

    def isExistUserName(userName)
        return getNameArray().include?(userName)
    end

    # ユーザー一覧から、ユーザー名が前方一致したものを返却。
    # 最初にヒットしたものを返す。見つからなければnilを返す。
    # TODO 複数ヒットしたらエラーにできるように
    def getUserNameWithForwardMatch(userNamePart)
        return "all" if userNamePart == "all"

        @userList.each do |user|
            return user.userName if user.userName.start_with?(userNamePart)
        end
        return nil
    end

end